### PR TITLE
Adding ServiceMonitor to ML Pipelines

### PR DESCRIPTION
**Pull request by DharmitD**
## How Has This Been Tested?

- Ran the pipeline tests without running the functions to delete pipelines and runs.
- Curled to the `ml-pipelines` service and listed out the pipeline metrics: http://pastebin.test.redhat.com/1068121
- Applied the ServiceMonitor config `monitor.yaml` 
- Ran the following command querying one of the metrics to see if it returns the expected value:
```
$ oc -n openshift-monitoring exec -c prometheus prometheus-k8s-0 -- curl -k -H "Authorization: Bearer $monitoring_token" 'https://thanos-querier.openshift-monitoring.svc:9091/api/v1/query?query=pipeline_server_get_requests'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0{"status":"success","data":{"resultType":"vector","result":[{"metric":{"__name__":"pipeline_server_get_requests","container":"ml-pipeline-api-server","endpoint":"http","instance":"10.128.3.27:8888","job":"ml-pipeline","namespace":"opendatahub","pod":"ml-pipeline-55fc9dbbc4-dk2h2","prometheus":"openshift-user-workload-monitoring/user-workload","service":"ml-pipeline"},"value":[1659121271.337,"7"]}]}}
100   403  100   403    0     0  20150      0 --:--:-- --:--:-- --:--:-- 20150
```

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: kfp-tekton/prometheus/monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    application-crd-id: kubeflow-pipelines
+  name: ml-pipelines-metrics-monitor
+spec:
+  endpoints:
+    - path: /metrics
+      port: http
+  selector:
+    matchLabels:
+      application-crd-id: kubeflow-pipelines
+
```
Comment (anishasthana) [inline kfp-tekton/prometheus/monitor.yaml:14]: Something went very wrong here

**Review by anishasthana — CHANGES_REQUESTED** [review]
We would also need the kustomization.yaml file that is above this to actually call it. As it stands right now, the service monitor will never be deployed

File: kfp-tekton/prometheus/monitor.yaml (hunk 1 above)
Comment (DharmitD) [reply kfp-tekton/prometheus/monitor.yaml]: Does it look fine now?

**Comment by DharmitD** [comment]
> We would also need the kustomization.yaml file that is above this to actually call it. As it stands right now, the service monitor will never be deployed

The `kustomization.yaml` is added to the commit: `kfp-tekton/prometheus/kustomization.yaml`. It's calling the `monitor.yaml`.
Do I need to add anything more to it?

File: kfp-tekton/prometheus/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+resources:
```
Comment (anishasthana) [inline kfp-tekton/prometheus/kustomization.yaml:1]: imo the kustomization file in `kfp-tekton/base` should have a reference to this folder too.

**Review by anishasthana — CHANGES_REQUESTED** [review]

File: kfp-tekton/prometheus/kustomization.yaml (hunk 2 above)
Comment (DharmitD) [reply kfp-tekton/prometheus/kustomization.yaml:1]: updated the kustomization file in `kfp-tekton/base` to have a reference to this folder.

**Review by anishasthana — COMMENTED** [review]
/lgtm

**Review by crobby — APPROVED** [review]
/lgtm

**Review by crobby — APPROVED** [review]
/lgtm

**Review by anishasthana — APPROVED** [review]
/lgtm

**Pull request merged**